library/spdm_responder_lib/libspdm_rsp_key_exchange.c

```diff
@@ -362,10 +362,16 @@ libspdm_return_t libspdm_get_response_key_exchange(void *context,
     }
     if (spdm_response->mut_auth_requested != 0) {
 #if (LIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP) || (LIBSPDM_ENABLE_CAPABILITY_ENCAP_CAP)
-        libspdm_init_mut_auth_encap_state(
-            context, spdm_response->mut_auth_requested);
-        spdm_response->req_slot_id_param =
-            (spdm_context->encap_context.req_slot_id & 0xF);
+        if (spdm_response->mut_auth_requested == SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED) {
+            /* no need to libspdm_init_mut_auth_encap_state() because of no ENCAP message */
+            spdm_response->req_slot_id_param =
+                (spdm_context->encap_context.req_slot_id & 0xF);
+        } else {
+            libspdm_init_mut_auth_encap_state(
```

Comment:
Would you please confirm if this should be updated as well?
As far as I can see, the possible sequences should be only
`     *    2.1 GET_DIGEST/GET_CERTIFICATE (spdm_context->encap_context.req_slot_id != 0xFF) */`
`     *    2.2 N/A (SPDM_GET_CAPABILITIES_REQUEST_FLAGS_PUB_KEY_ID_CAP, spdm_context->encap_context.req_slot_id must be 0xFF)`

https://github.com/DMTF/libspdm/blob/befd3a8c41427feb28719db4324ec2f82ed62790/library/spdm_responder_lib/libspdm_rsp_encap_response.c#L199-L226

Reply:
you are right. Will update as well.

Reply:
@xiangfeima, please review the update if possible.